### PR TITLE
Merge/sharedaddy divergence

### PR DESCRIPTION
**Pull request by dereksmart**
These files were divergent from dotcom. This updates our repo to include all of their changes and we'll merge our changes back to wpcom when these are approved and merged.

### CODE REVIEW

File: modules/sharedaddy/sharing.js [hunk 1]
```diff
@@ -42,8 +42,12 @@ if ( sharing_js_options && sharing_js_options.counts ) {
 						continue;
 					}
 
-					while ( ( service_url = urls[ service ].pop() ) ) {
-						jQuery.getScript( service_url );
+					while ( ( service_request = requests[ service ].pop() ) ) {
```
Comment (aduth) [inline modules/sharedaddy/sharing.js]: These changes can be omitted, since Twitter requests were the only non-string requests.

Comment (aduth) [inline modules/sharedaddy/sharing.js]: They also introduce an error in `sharing.js` when tested on my Jetpack site:

`Uncaught ReferenceError: requests is not defined`

I think a few lines from the source commit were omitted (specifically, the renaming of some variables). Those should probably be migrated here to simplify future syncing.

**Comment by dereksmart** [comment]
@aduth -- updated to use `requests` throughout. look ok?

File: modules/sharedaddy/sharing.js [hunk 2]
```diff
@@ -4,7 +4,7 @@ if ( sharing_js_options && sharing_js_options.counts ) {
 	var WPCOMSharing = {
 		done_urls : [],
 		get_counts : function() {
-			var url, urls, id, service, service_url;
+			var https_url, http_url, url, requests, id, service, service_request;
```
Comment (aduth) [inline modules/sharedaddy/sharing.js]: `https_url` and `http_url` were removed in b0b68a2c6b62f627191d567759234153e1e1d463 and shouldn't be re-added here.

File: modules/sharedaddy/sharing.js [hunk 3]
```diff
@@ -37,13 +37,17 @@ if ( sharing_js_options && sharing_js_options.counts ) {
...
-				for ( service in urls ) {
+				for ( service in requests ) {
 					if ( ! jQuery( 'a[data-shared=sharing-' + service + '-' + id  + ']' ).length ) {
 						continue;
 					}
 
-					while ( ( service_url = urls[ service ].pop() ) ) {
-						jQuery.getScript( service_url );
+					while ( ( service_request = requests[ service ].pop() ) ) {
+						if ( 'string' === typeof service_request ) {
```
Comment (aduth) [inline modules/sharedaddy/sharing.js]: It's not hurting anything, but this condition ought to be removed, i.e. only use `jQuery.getScript` ([as noted in an earlier comment](https://github.com/Automattic/jetpack/pull/2816/files#r41426366)). If you'd prefer, we can wait until this is removed from the WPCOM side.

**Comment by dereksmart** [comment]
Updated again & removed unnecessary conditional

**Comment by aduth** [comment]
Looking good :+1:

**Pull request merged**